client-js/app/scripts/app.ts

```diff
@@ -141,13 +141,20 @@ window.addEventListener('WebComponentsReady', () => {
 
   class RegistryActivity implements Activity {
     path: Array<string>;
+    private globElem: LabradRegistry;
 
-    constructor(path: Array<string>) {
+    private api: registry.RegistryServiceJsonRpc;
+    private lifetime = new Lifetime();
+
+    constructor(api: registry.RegistryServiceJsonRpc, path: Array<string>) {
+      this.api = api;
       this.path = path;
     }
 
     start(): Promise<ActivityState> {
+      this.api.watch({path: this.path});
```

Comment:
It's definitely a bit awkward to string together promises like this (arrow functions help, but don't alleviate all the pain). The next version of typescript, 1.7, will have async/await, which will make it possible to do inlineCallbacks-style coroutines instead of writing chains of callbacks like this; it can't come soon enough, IMO (The typescript roadmap is [here](https://github.com/Microsoft/TypeScript/wiki/Roadmap)). Until then, we have to just suffer through the pain.